query/src/main/java/org/infinispan/query/dsl/embedded/impl/HibernateSearchPropertyHelper.java

```diff
@@ -87,6 +88,15 @@ public Class<?> getIndexedPropertyType(Class<?> entityType, String[] propertyPat
       return fieldDescriptor.type().dslArgumentClass();
    }
 
+   @Override
+   public boolean isNestedIndexStructure(Class<?> entityType, String[] propertyPath) {
+      IndexFieldDescriptor fieldDescriptor = getFieldDescriptor(entityType, propertyPath);
+      if (fieldDescriptor == null || fieldDescriptor.type() == null) {
+         return false;
+      }
+      return fieldDescriptor.type().traits().contains(IndexFieldTraits.Predicates.NESTED);
```

Comment:
It seems that the problem is the call of the method `org.hibernate.search.engine.backend.metamodel.IndexFieldDescriptor.type()`, since `HibernateSearchPropertyHelper` is linked at GraalVM image build time, probably we cannot use this method.
I think that we need to add the information from the metadata Hibernate Search API.
Maybe we can include the feature here => https://hibernate.atlassian.net/browse/HSEARCH-3909 check it

You can for the moment adding a TODO:
``` java
IndexFieldDescriptor fieldDescriptor = getFieldDescriptor(entityType, propertyPath);

      // TODO https://hibernate.atlassian.net/browse/HSEARCH-3909 check it
      return true;
```

With this change it seems that the build passes.

Reply:
Thank you !!
I'll add the TODO and then check if there are any broken tests.